src/main/java/net/staticsnow/nexus/repository/apt/AptUploadHandler.java

```diff
@@ -61,26 +61,37 @@ public AptUploadHandler(@Named("simple") final VariableResolverAdapter variableR
   public UploadResponse handle(Repository repository, ComponentUpload upload) throws IOException {
     AptHostedFacet hostedFacet = repository.facet(AptHostedFacet.class);
     StorageFacet storageFacet = repository.facet(StorageFacet.class);
-    return TransactionalStoreBlob.operation.withDb(storageFacet.txSupplier()).throwing(IOException.class).call(() -> {
-      try (TempBlob tempBlob = storageFacet.createTempBlob(upload.getAssetUploads().get(0).getPayload(), FacetHelper.hashAlgorithms)) {
-        
-        ControlFile controlFile = AptPackageParser.parsePackage(tempBlob);
-        if (controlFile == null) {
-          throw new IOException("Invalid debian package:  no control file");
-        }
-        String name = controlFile.getField("Package").map(f -> f.value).get();
-        String version = controlFile.getField("Version").map(f -> f.value).get();
-        String architecture = controlFile.getField("Architecture").map(f -> f.value).get();
-        String assetPath = FacetHelper.buildAssetPath(name, version, architecture);
-        ensurePermitted(repository.getName(), AptFormat.NAME, assetPath, Collections.emptyMap());
-        try {
-          Asset asset = hostedFacet.ingestAsset(upload.getAssetUploads().get(0).getPayload());
-          return new UploadResponse(asset);
-        } catch (PGPException e) {
-          throw new IOException(e);
-        }
+
+    try (TempBlob tempBlob = storageFacet.createTempBlob(upload.getAssetUploads().get(0).getPayload(), FacetHelper.hashAlgorithms)) {
+      ControlFile controlFile = AptPackageParser.parsePackage(tempBlob);
+      if (controlFile == null) {
+        throw new IOException("Invalid debian package:  no control file");
+      }
+      String name = controlFile.getField("Package").map(f -> f.value).get();
+      String version = controlFile.getField("Version").map(f -> f.value).get();
+      String architecture = controlFile.getField("Architecture").map(f -> f.value).get();
+      String assetPath = FacetHelper.buildAssetPath(name, version, architecture);
+
+      doValidation(repository, assetPath);
+
+      UnitOfWork.begin(storageFacet.txSupplier());
+      try {
+        Asset asset = hostedFacet.ingestAsset(upload.getAssetUploads().get(0).getPayload());
```

Comment:
Is there already a transactional context here? As we no longer explicitly start the transaction, I was expecting an annotation on the method to do so for us.